src/blueapi/core/context.py

```diff
@@ -198,6 +198,13 @@ def valid(cls, value):
                         raise ValueError(f"Device {value} is not of type {target}")
                     return val
 
+                @classmethod
+                def __modify_schema__(
+                    cls, field_schema: dict[str, Any], field: Optional[ModelField]
+                ):
+                    if field:
+                        field_schema.update({field.name: repr(target)})
+
```

Comment:
Is this needed because pydantic just doesn't make a schema by default if it isn't there?

Reply:
Yes, with allow_arbitrary_types, and types that aren't primitives or Pydantic models, we have to provide a serialisation method